rpc/provider.go

```diff
@@ -192,6 +201,12 @@ type RPCProvider interface {
 	TransactionStatus(ctx context.Context, transactionHash *felt.Felt) (*TxnStatusResult, error)
 }
 
+// RPCUtils is the interface that defines the RPC utility methods.
+type RPCUtils interface {
+	IsCompatible(ctx context.Context) (bool, string, error)
+	EstimateTip(ctx context.Context) (U64, error)
```

Comment:
This also a specific implementatino detail of your type, I don't think it should be an interface either